flood_forecast/evaluator.py

```diff
@@ -106,18 +121,32 @@ def infer_on_torch_model(
             or no columns if num_prediction_samples is None
     """
     device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
-    if type(datetime_start) == str:
+    if isinstance(datetime_start, str):
```

Comment:
Are instances better?

Reply:
in general isinstance is a bit nicer than type(x) == type(y) since isinstance takes care of subclasses etc.